server/src/com/cloud/dc/dao/DedicatedResourceDaoImpl.java

```diff
@@ -312,6 +317,65 @@ public DedicatedResourceVO findByHostId(Long hostId) {
     }
 
     @Override
+    public List<DedicatedResourceVO> listAvailableResources(Long accountId, Long... domains) {
```

Comment:
@koushik-das Could not find a way to write the conditional statement so went with this way. Here account_id in involved in two places in condition.